packages/@aws-cdk/pipelines/lib/blueprint/pipeline-queries.ts

```diff
@@ -0,0 +1,67 @@
+import { Step, ScriptStep, StackOutputReference, StackDeployment, StackAsset, StageDeployment } from '.';
+import { Pipeline } from '../main/pipeline';
+
+/**
+ * Answer some questions about a pipeline blueprint
+ */
+export class PipelineQueries {
+  constructor(private readonly pipeline: Pipeline) {
+  }
+
+  /**
+   * Return the names of all outputs for the given stack that are referenced in this blueprint
+   */
+  public stackOutputsReferenced(stack: StackDeployment): string[] {
```

Comment:
Could we just move these to Pipeline either directly or under pipeline.queries.xxxx

Reply:
I'm not going to. 

I prefer to keep the Pipeline, Wave, StageDeployment etc classes as simple data holders; we can then add on as many helpers and queries and whatnot as we need, without needing to change anything there.

Reply:
OK. Legit.